### ISSUE TITLE
Regression using `role` identifier

### ISSUE THREAD

**Issue by asr** [ndmitchell/hlint#196]
Running HLint 1.9.23-25 on

```haskell
foo :: Int -> Int
foo role = 42
```

generates the error

```
Warning: Parse error: role
Found:
    foo :: Int -> Int
    foo role = 42
```
 
It works with Hlint 1.9.22 through.

**Comment by mpickering** [comment]
I am handling this @ndmitchell

**Comment by mpickering** [comment]
I think this is because hlint turns on the "RoleAnnotations" extension which steals the `role` keyword. The file parses fine in `haskell-src-exts`.

**Comment by ndmitchell** [comment]
I've raised https://github.com/haskell-suite/haskell-src-exts/issues/286 to track it. GHC doesn't steal the `role` keyword everywhere, so this code is valid GHC but not valid HSE.

@asr - in the meantime just run with `-XNoRoleAnnotations` and it will continue to parse.

**Comment by asr** [comment]
> @asr - in the meantime just run with `-XNoRoleAnnotations` and it will continue to parse.

Thanks, it worked.

**Comment by mpickering** [comment]
I also fixed the bug in HSE fwiw.

**Comment by ndmitchell** [comment]
Thanks @mpickering. Is this (and presumably any future HSE changes) going to sit in the repo for a year, or get gradually released as they are made? Not sure what HSE intends to do for minor but hits a few people style bugs?

**Comment by mpickering** [comment]
I don't object to doing more regular releases but it seems that a lot of people put very tight bounds on HSE which means a lot of packages get broken whenever I do a release.

**Comment by ndmitchell** [comment]
This is a no API change though. More generally I'd be fine new point release that only add new ADT values, although you are still going to have to break every so often.

**Comment by mpickering** [comment]
Right but there is a non-negligible cost to maintaining two branches which I would rather avoid.

**Comment by asr** [comment]
+1 for releasing bug-fixed versions of HSE more frequently.

**Comment by ndmitchell** [comment]
I'd be fine with an accelerated release cycle. I appreciate two branches are a hassle, its something I certainly avoid.

**Comment by asr** [comment]
The issue was fixed on HLint 1.9.36.

**Comment by ndmitchell** [comment]
Thanks for spotting! I still need to go through the tickets.
